extensions/oidc-common/runtime/src/main/java/io/quarkus/oidc/common/OidcRequestContextProperties.java

```diff
@@ -19,22 +18,55 @@ public OidcRequestContextProperties(Map<String, Object> properties) {
         this.properties = properties;
     }
 
+    /**
+     * Get property value
+     *
+     * @param name property name
+     * @return property value
+     */
     public <T> T get(String name) {
         @SuppressWarnings("unchecked")
         T value = (T) properties.get(name);
         return value;
     }
 
+    /**
+     * Get property value as String
+     *
+     * @param name property name
+     * @return property value as String
+     */
     public String getString(String name) {
         return (String) get(name);
     }
 
+    /**
+     * Get typed property value
+     *
+     * @param name property name
+     * @param type property type
+     * @return typed property value
+     */
     public <T> T get(String name, Class<T> type) {
         return type.cast(get(name));
     }
 
+    /**
+     * Get all properties
+     *
+     * @return all properties
+     */
     public Map<String, Object> getAll() {
-        return Collections.unmodifiableMap(properties);
+        return properties;
```

Comment:
Or remove the `set` method and let users change the Map freely, your choice

Reply:
Removing the `set` may be a better choice, given that you might also have to support `remove`, `addAll`, etc

Reply:
@gastaldi Hi George,  I'd be inclined to have shortcuts like `set`, though may  be it is better be called `put`, to let users avoid having to go via Map indirection...
I should've avoided adding that getAll,
But now, would it really make any better to keep the map view unmodifiable ?
Users can choose a shortcut or Map way otherwise

Reply:
Right, `put` sounds better, but if you're returning a modifiable Map, why bother changing the API?